### DISCUSSION TITLE
MySQL Over Local Network

### DISCUSSION THREAD

**Discussion by ulfie22** [tighten/takeout#270] [Q&A] [answered]
Hi Takeout Fans -

Is there an easy way to make my local MySQL database available over my local network (or, I guess, over the Internet). This would be a big help in sharing a development database. Looking at the Docker app, it seems to be bound to 0.0.0.0::3307 - is there a way to change that (through Takeout or through the Docker app) to the actual IP address (NAT or real) of the machine?

Thanks!!

**Comment by josecanhelp** [comment, marked as answer]
Hey @ulfie22!

Takeout does not have a built-in network tunnel to share connections externally, but you can easily use [ngrok](https://ngrok.com/) to do that.

Install ngrok and run `ngrok tcp 3306` (or whatever port your takeout container is bound to). This will provide a forwarding address that you can share with external networks.

<img width="781" alt="Screen Shot 2022-01-11 at 12 22 46 PM" src="https://user-images.githubusercontent.com/2329654/149015697-1e441b29-64aa-4870-8231-7edd9051b10f.png">

For example, in the screenshot above, you can see a forwarding address of `tcp://8.tcp.ngrok.io:15423`

From an external network, you can connect via mysql to host `8.tcp.ngrok.io` with port `15423`.

**Reply by ulfie22** [reply]
Wow @josecanhelp !! That worked great !! Thank you so much!!
